src/System.IO.FileSystem/src/System/IO/CharSpanExtensions.Windows.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace System.IO
+{
+    internal static partial class CharSpanExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Instead of adding AggressiveInlining inlining that leads to codebloat, you should avoid the need for marshaling in  CompareStringOrdinal (use char* and BOOL in the signature) and do the marshaling yourself (using fixed, etc.). It will be smaller and faster code that way.

Reply:
I'm not sure I follow exactly. Do fully blittable arguments (when they're simple types) make that much of a difference in imports? We've been encouraging using ref char in other places to avoid needing to use fixed... @stephentoub I recall you suggesting ref char for Span<char>- I might be remembering incorrectly. I want to make sure we're clear on guidance here.

The results and error handling aren't exactly straight forward so I think having a wrapper method for usage is appropriate. (I also think this is a method we'd want to ultimately have public). I'm fine with removing the AggressiveInlining if the advantage is trivial.

Reply:
I've encouraged using ref instead of pointers to keep the code simpler and minimize unsafe pointer usage that we could mess up.  If specific cases benefit from using pointers instead for perf, that's fine.